### ISSUE TITLE
Bug: Initial state can't be restored for machines with transitioned child machines  

### ISSUE THREAD

**Issue by faustAbc** [statelyai/xstate#3547]
### Description

Let's say we have root machine that creates child machines and transits their state.
If we serialize this state, deserialize it, and then try to use useMachine hook from @xstate/react with options.state set to deserialized state, we will get the following error:
```
Unable to send event to child 'Child id' from service '(machine)'.
```

### Expected result

No errors, parent and child stated are deserialized correctly

### Actual result

The error is emitted

### Reproduction

https://codesandbox.io/s/keen-nobel-i8mymo?file=/src/App.tsx

### Additional context

Steps to reproduce the issue in the codesandbox:
1. Press 'Spawn child'
2. Press 'Move children to second state'
3. Reload the sandboxed page
4. Expect error

**Comment by davidkpiano** [comment]
Recursive persistence is addressed here: #3743

**Comment by davidkpiano** [comment]
This should now be working as expected in `xstate@beta`
